### PR TITLE
An attempt to remove dependencies of Self-Replicating Robots

### PR DESCRIPTION
**Pull request by EthanDobbs**
This is not quite ready to merge, as you currently can't see what cards are attached to an opponent's instance of SRR without digging through the game logs. This PR was opened so others can suggest changes.

SRR appears in many places of the code where it shouldn't. If I want to modify what happens when you play a card, I shouldn't have to worry about breaking SRR. The goal of this PR is to improve maintainability in regards to the many dependencies of SRR by relocating relevant code to the SRR file.

This PR will change card serialization, so implementing this code on your server will break existing games.

### CODE REVIEW

File: src/client/components/OtherPlayer.vue [hunk 1]
```diff
@@ -21,14 +21,6 @@
                 <stacked-cards :cards="getCardsByType(player.tableau, [CardType.EVENT])" :player="player"></stacked-cards>
             </div>
         </div>
-        <div v-if="player.selfReplicatingRobotsCards.length > 0" class="player_home_block">
```
Comment (kberg) [inline src/client/components/OtherPlayer.vue:24]: Why is this better? SRR cards should really be visually separate from other cards.

File: src/server/cards/promo/SelfReplicatingRobots.ts [hunk 2]
```diff
@@ -8,18 +8,22 @@ import {SelectCard} from '../../inputs/SelectCard';
...
+import {CardModel} from '../../../common/models/CardModel';
+import {newProjectCard} from '../../../server/createCard';
 
 export class SelfReplicatingRobots extends Card implements IProjectCard {
   constructor() {
     super({
       type: CardType.ACTIVE,
       name: CardName.SELF_REPLICATING_ROBOTS,
       cost: 7,
+      resourceType: CardResource.RESOURCE_CUBE,
```
Comment (kberg) [inline src/server/cards/promo/SelfReplicatingRobots.ts:26]: This will make SRR susceptible to cards like CEO's favorite project, etc.

**Comment by kberg** [comment]
What's the fundamental mechanical change? Can you describe it? It will help guide me through the review.

(I bet there's a way to preserve existing games, let's see.)

File: src/client/components/OtherPlayer.vue (hunk 1 above)
Comment (EthanDobbs) [reply src/client/components/OtherPlayer.vue:24]: This is part of the PR that isn't finalized yet and why I changed it to a draft.

File: src/server/cards/promo/SelfReplicatingRobots.ts (hunk 2 above)
Comment (EthanDobbs) [reply src/server/cards/promo/SelfReplicatingRobots.ts:26]: True, this is intentional.

**Pull request closed without merging**